target_postgres/postgres.py

```diff
@@ -582,36 +559,51 @@ def persist_rows(self,
         default_fields = {k: v.get('default') for k, v in target_table_json_schema['properties'].items()
                           if v.get('default') is not None}
 
-        mapped_fields = target_schema.get('mappings', {})
-        mapped_from_fields = set([v['from'] for k,v in mapped_fields.items()])
-
-        rows = iter(records)
+        records_iter = iter(records)
 
         def transform():
             try:
-                row = next(rows)
-                with io.StringIO() as out:
-                    for prop in headers:
-                        if prop in mapped_fields:
-                            row[prop] = RESERVED_NULL_DEFAULT
-                        else:
-                            row[prop] = self.serialized_value(prop,
-                                                              row.get(prop, None),
-                                                              default_fields,
-                                                              datetime_fields)
-
-                    for prop in mapped_from_fields:
-                        if prop in row:
-                            mapped_field = self.get_mapping(target_schema,
-                                                            prop,
-                                                            target_table_json_schema['properties'][prop])
-
-                            row[mapped_field] = self.serialized_value(prop,
-                                                                      row.get(prop, None),
-                                                                      default_fields,
-                                                                      datetime_fields)
-                            row.pop(prop, None)
+                record = next(records_iter)
+                row = {}
+
+                for field in set(headers + list(record.keys())):
```

Comment:
If we are validating JSON, shouldn't `headers` and `record.keys()` be the same? If there are extra columns in `record.keys()`, wouldn't that fail?

Reply:
Column splitting. Headers will contain the mapped fields ie `name__s`, whereas record will contain `name` etc. I can also get this from filtering the mapped field names. Lemme do that instead to make this clearer.